src/clincoded/static/components/dashboard.js

```diff
@@ -360,84 +551,78 @@ var Dashboard = createReactClass({
-                </div>
-                <div className="panel-content-wrapper">
-                    {this.state.vciInterpListLoading ? showActivityIndicator('Loading... ') : null}
-                    {records.length > 0 ?
-                        <table className="table individual-interpretation-list">
-                            <thead>
-                                <tr>
-                                    <th className="item-variant">Variant</th>
-                                    <th className="item-attribute">Disease/Mode of Inheritance</th>
-                                    <th className="item-status">Status</th>
-                                    <th className="item-timestamp">Creation Date</th>
-                                </tr>
-                            </thead>
-                            <tbody>
-                                {records.map(item => {
-                                    return (
-                                        <tr key={item.uuid}>
-                                            <td className="item-variant">
-                                                <a key={item.uuid}
-                                                    className="individual-record-link"
-                                                    href={"/variant-central/?edit=true&variant=" + item.variantUuid + "&interpretation=" + item.uuid}>
-                                                    <span className="variant-title"><strong>{renderVariantTitle(item.variant)}</strong></span>
-                                                </a>
-                                            </td>
-                                            <td className="item-attribute">{item.diseaseTerm ? item.diseaseTerm : "--"}/{item.modeInheritance ? item.modeInheritance : "--"}</td>
-                                            <td className="item-status">{self.renderClassificationStatusTag(null, item.interpretation, this.props.affiliation, this.props.session)}</td>
-                                            <td className="item-timestamp">{moment(item.date_created).format("YYYY MMM DD, h:mm a")}</td>
-                                        </tr>
-                                    );
-                                })}
-                            </tbody>
-                        </table>
-                        :
-                        <div className="panel-body"><p>You have not created any variant interpretations.</p></div>
-                    }
+    renderInterpretationsStatusFilter(listLength) {
+        if (listLength > 0) {
+            const statusList = [
+                {value: "None", label: "None"},
+                {value: "In Progress", label: "In Progress"},
+                {value: "Approved", label: "Approved"},
+                {value: "Provisional", label: "Provisional"},
+                {value: "New Provisional", label: "New Provisional"},
+                {value: "Published", label: "Published"},
+               ];
```

Comment:
Made above changes

Reply:
@h-tong , I think it's better not to create new functions to get the "stringOnly" result so we don't have to maintain two copies of similar logic.  Also the same "stringOnly" logic is already added to "renderVariantTitle" in render_variant_title.js.  If you think the word "render" has special meaning, we can rename the function calls and file names.

Reply:
I was thinking that we should extract the common logic from these "render" functions and make a utility function, not necessarily duplicating the logic. I think the issue isn't just with the name of the function. There is an inconsistent return type, so it's not always clear whether you're going to get a string or a set of React elements as the return value. After looking at it more closely, it looks like it could be quite a large refactor, so I guess it's okay to just leave it since it works. Maybe it could be helpful to have a discussion with Mark and Bryan about this when we're a little less pressed for time.